main.py

```diff
@@ -1511,43 +1512,90 @@ def show_hover(self, point, contents):
             max_width=800)
 
 
-class CompletionHandler(sublime_plugin.EventListener):
-    def __init__(self):
-        self.completions = []  # type: List[Tuple[str, str]]
-        self.refreshing = False
+class CompletionState(object):
+    IDLE = 0
+    REQUESTING = 1
+    APPLYING = 2
+    CANCELLING = 3
 
-    def on_query_completions(self, view, prefix, locations):
-        if not is_supported_view(view):
-            return None
 
-        if not self.refreshing:
-            client = client_for_view(view)
+class CompletionHandler2(sublime_plugin.ViewEventListener):
+    def __init__(self, view):
+        self.view = view
+        self.autocomplete_triggers = None
+        self.completions = []  # type: List[Tuple[str, str]]
+        self.state = CompletionState.IDLE
+        self.next_request = None
 
-            if not client:
-                return
+    @classmethod
+    def is_applicable(cls, settings):
+        syntax = settings.get('syntax')
```

Comment:
In rare cases `syntax` may be `None` (annoying ST bug), check for this just in case.

Reply:
In this case the behaviour would be as expected, but I'll add this fix anyways.
Is this documented somewhere?